### PR TITLE
Accessibility, keyboard support

### LINKED ISSUE #57: Implement accessibility based on ARIA practices
Hello 👋   
I've been thinking about making the color picker accessible, and I would like to discuss the related changes.

It looks like for `Hue` and `Alpha` sliders we can follow [WAI-ARIA authoring practices](https://www.w3.org/TR/wai-aria-practices/#slider). See also the [example](https://www.w3.org/TR/wai-aria-practices/examples/slider/slider-1.html).
I'm not an expert, but presumably the same approach should work for `Saturation` as well.

There are following open questions that come to my mind:

1. Implementing this would require to figure out what the proper `step` value per arrow key press should be.
Should we make this step configurable? Maybe not as pubic property, but a protected one.

2. Note that `Page Up` and `Page Down` allow to change value by multiple steps (10 steps in the example).
Should this be configurable as well?

3. Proper support for screen reader announcements would require to provide a way for setting `aria-label` on the pointer elements. It could be done either with individual props or by passing an object with the following structure:

```js
i18n: {
  alpha: 'Alpha',
  hue: 'Hue',
  saturation: 'Saturation'
}
```

I'm going to prototype some of these changes in `vanilla-colorful` so I will create an issue there as well.

### PR DESCRIPTION
**Pull request by omgovich**
Closes #57

### CODE REVIEW

**Comment by omgovich** [comment]
Had to apply a few tricks to keep all pickers inside 2 KB size limit.

**Comment by omgovich** [comment]
@molefrog @rschristian Hey guys! Could I ask you to take a look?

**Comment by omgovich** [comment]
HexColorPicker size:
```
1.51 → 1.86 KB
```

I added a lot of new code, so I think it's a pretty good result.

File: src/utils/limit.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Limit number within bounds.
+// Used ternary operator instead of `Math.min(Math.max(min, number), max)` to save few bytes
+export const limit = (number: number, min = 0, max = 1): number => {
```
Comment (molefrog) [inline src/utils/limit.ts:3]: Imo, there is a better name for this method: `clamp`, used widely in animations (react-spring) and [the latest CSS proposal](https://developer.mozilla.org/en-US/docs/Web/CSS/clamp).

Comment (omgovich) [reply src/utils/limit.ts:3]: Agree

Comment (omgovich) [reply src/utils/limit.ts]: Done in 99be541

**Comment by omgovich** [comment]
Added few specs to test keyboard support.

File: src/utils/clamp.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Clamps a value between an upper and lower bound.
+// Used ternary operator instead of `Math.min(Math.max(min, number), max)` to save few bytes
+export const clamp = (number: number, min = 0, max = 1): number => {
+  return number > max ? max : number < min ? min : number;
```
Comment (rschristian) [inline src/utils/clamp.ts:4]:
```suggestion
  return Math.max(min, Math.min(number, max));
```

FWIW, I think this is like ~10-20b smaller. Counting pennies, but personally I think it's more readable too.

File: src/components/common/Alpha.tsx [hunk 3]
```diff
@@ -13,16 +16,19 @@ interface Props {
...
-    [onChange]
-  );
+  // Alpha always fit into [0, 1] range
+  const handleMove = (interaction: Interaction) => {
+    onChange({ a: interaction.left });
+  };
+
+  const handleKey = (offset: Interaction) => {
+    onChange({ a: clamp(hsva.a + offset.left) });
+  };
```
Comment (rschristian) [inline src/components/common/Alpha.tsx:26]: Should the comment (`// Alpha always...`) be moved down to be above the usage of `clamp()`?

File: src/utils/convert.ts [hunk 4]
```diff
@@ -32,7 +32,7 @@ export const hslaStringToHsva = (hslString: string): HsvaColor => {
     h: Number(match[1]),
     s: Number(match[2]),
     l: Number(match[3]),
-    a: match[4] !== undefined ? Number(match[4]) : 1,
+    a: match[4] == null ? 1 : Number(match[4]),
```
[the remark this answers is not in the document]
Comment (omgovich) [reply src/utils/convert.ts:35]: It's just shorter version of `x === undefined`. Saves few bites.

See:
image

File: src/utils/clamp.ts (hunk 2 above)
Comment (omgovich) [reply src/utils/clamp.ts:4]: In this project, the output size is more important than the source code style. 
We use tricks and they make code worse-looking but we cover then with comments.

File: src/utils/convert.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (omgovich) [reply src/utils/convert.ts]: But you're right it's too confusing. I returned the previous version (with `=== undefined`).

File: src/utils/clamp.ts (hunk 2 above)
Comment (rschristian) [reply src/utils/clamp.ts:4]: Yes, what I'm saying is that the suggestion I provided is 10-20b lighter. 

[Existing outputs to 207 bytes](https://babeljs.io/repl#?browsers=IE%2011&build=&builtIns=usage&spec=false&loose=true&code_lz=MYewdgzgLgBMA2BDAtgBxgXhgCjAV2QCMBTAJwBoZkBLMTGABkuUQA96BGASkwD4YA3gCgYMUsSh5SdALKIoACwB0LVthphKcxStq4CJClTZcuAbiEBfIUA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=true&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Cenv&prettier=false&targets=&version=7.11.6&externalPlugins=)

[My suggestion outputs to 195](https://babeljs.io/repl#?browsers=IE%2011&build=&builtIns=usage&spec=false&loose=true&code_lz=MYewdgzgLgBMA2BDAtgBxgXhgCjAV2QCMBTAJwBoZkBLMTGABkuUQA96BGASkwD4YA3gCgYMUsSh5SdALKIoACwB0LVthphKcxStq4CJClTZcuAbiEBfIUA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=true&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Cenv&prettier=false&targets=&version=7.11.6&externalPlugins=)

Comment (rschristian) [reply src/utils/clamp.ts:4]: So you shave off 12b using `Math.max()` and the like

Comment (omgovich) [reply src/utils/clamp.ts:4]: Got you. But you should compare minified code. Solution based on ternary operators is almost 2 times shorter.
```js
Math.max(i,Math.min(n,a));
n>a?a:n<i?i:n;
```

Comment (rschristian) [reply src/utils/clamp.ts:4]: Ah, true enough.

Comment (omgovich) [reply src/utils/clamp.ts:5]: Described this thing in 6e810e6

Comment (rschristian) [reply src/utils/clamp.ts:5]: Cheers. Sorry about that.

**Review by molefrog — APPROVED** [review]
Great work!

**Comment by omgovich** [comment]
Thanks, @molefrog @rschristian!

**Pull request merged**